### DISCUSSION TITLE
Update some deprecated actions being used by trunk-action

### DISCUSSION THREAD

**Discussion by bbush-uthealth** [trunk-io/docs#357] [Feature Requests]
trunk-action uses a few GitHub action versions that are deprecated and are triggering warnings in our GitHub workflows.

https://github.blog/changelog/2024-03-07-github-actions-all-actions-will-run-on-node20-instead-of-node16-by-default/

https://github.blog/changelog/2024-04-16-deprecation-notice-v3-of-the-artifact-actions/

---

In https://github.com/trunk-io/trunk-action/blob/main/action.yaml
Line 361 stanza is using a deprecated actions/upload-artifact:
    - name: Upload annotations artifact
      if: always() && env.TRUNK_UPLOAD_ANNOTATIONS == 'true'
      uses: actions/upload-artifact@v3
      with:
        name: trunk-annotations
        path: ${{ env.TRUNK_TMPDIR }}/annotations.bin

upload-artifact needs to be updated to v4 in order to continue functioning after November 30, 2024.

---

And... line 278 stanza is using a deprecated actions/cache:
    - name: Cache Linters/Formatters
      if: env.TRUNK_CHECK_MODE != 'none' && env.INPUT_CACHE == 'true'
      uses: actions/cache@v3
      with:
        path: ${{ env.INPUT_CACHE_PATH }}
        key: ${{ env.INPUT_CACHE_KEY }}

This needs to be updated to actions/cache@v4 to avoid deprecated warnings.

**Comment by gewenyu99** [comment]
Hey, totally missed this somehow, but thank you so much for pointing it out! I'll get @det to take a look.

**Reply by gewenyu99** [reply]
We bumped this a bit ago:
https://github.com/trunk-io/trunk-action/commit/0c109cf103b1bf745bb0cd8ec19d00a4609527d8
